### PR TITLE
Fix connection string and port

### PR DESCRIPTION
**Pull request by eshleebien**
-Fix connection formatting for username and password
-Fix wrong port number

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,11 +19,14 @@ module.exports = {
...
         }
 
+        credentials = (config.user || '')
+            + ((config.user && config.password) ?
+                    (':' + config.password) : '');
+
         connection = mongo.db('mongodb://'
-            + config.user + ':'
-            + config.password + '@'
+            + ((credentials) ? (credentials + '@') : '')
```
Comment (ravenjohn) [inline index.js]: `+ (credentials ? (credentials + '@') : '')`

File: index.js [hunk 2]
```diff
@@ -18,11 +19,14 @@ module.exports = {
             return this.connections[key];
         }
 
+        credentials = (config.user || '')
```
Comment (ravenjohn) [inline index.js:22]:
``` javascript
credentials = (config.user || '')
    + ((config.user && config.password)
        ? (':' + config.password)
        : '');
```

File: index.js [hunk 3]
```diff
@@ -18,11 +19,14 @@ module.exports = {
...
+            + ((config.user && config.password) ?
+                    (':' + config.password) : '');
+
         connection = mongo.db('mongodb://'
-            + config.user + ':'
-            + config.password + '@'
+            + ((credentials) ? (credentials + '@') : '')
             + (config.host || '127.0.0.1') + ':'
-            + (config.port || 27107) + '/'
+            + (config.port || 27017) + '/'
```
Comment (ravenjohn) [inline index.js:30]: :+1:

File: index.js (hunk 2 above)
Comment (eshleebien) [inline index.js:22]: updated :)

File: index.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ module.exports = {
 
     open : function (config) {
         var connection,
+            credentials = '',
```
Comment (ravenjohn) [inline index.js]: no need for initial value

File: index.js [hunk 5]
```diff
@@ -7,7 +7,8 @@ module.exports = {
 
     open : function (config) {
         var connection,
-            key = '',
+            credentials,
+            key,
```
Comment (ravenjohn) [inline index.js]: sfsadfdadf

**Pull request merged**